patches/tModLoader/Terraria/GameContent/Drawing/TileDrawing.cs.patch

```diff
@@ -66,6 +84,25 @@
  			Texture2D result = TextureAssets.Tile[tile.type].Value;
  			int tileStyle = 0;
  			int num = tile.type;
+@@ -2799,7 +_,8 @@
+ 			}
+ 		}
+ 
+-		private static bool IsTileDangerous(Player localPlayer, Tile tileCache, ushort typeCache) {
++		//TML: Added x/y for hook compatibility, made public as it's API-like and matches Main.IsTileSpelunkable that way
++		public static bool IsTileDangerous(int tileX, int tileY, Player localPlayer, Tile tileCache, ushort typeCache) {
+ 			bool flag = false || typeCache == 135 || typeCache == 137 || typeCache == 138 || typeCache == 484 || typeCache == 141 || typeCache == 210 || typeCache == 442 || typeCache == 443 || typeCache == 444 || typeCache == 411 || typeCache == 485 || typeCache == 85;
```

Comment:
Same with IsTileSpelunkable - For API-like sanity, turn this exact method internal and create a public overload for it that does not have the tileCache and typeCache values, but instead gets them from the coordinates.